packages/editor/src/components/post-featured-image/index.js

```diff
@@ -29,49 +30,63 @@ function PostFeaturedImage( { currentPostId, featuredImageId, onUpdateImage, onR
 		if ( has( media, [ 'media_details', 'sizes', mediaSize ] ) ) {
 			mediaWidth = media.media_details.sizes[ mediaSize ].width;
 			mediaHeight = media.media_details.sizes[ mediaSize ].height;
 			mediaSourceUrl = media.media_details.sizes[ mediaSize ].source_url;
+			mediaAltText = media.media_details.sizes[ mediaSize ].alt_text;
 		} else {
 			mediaWidth = media.media_details.width;
 			mediaHeight = media.media_details.height;
 			mediaSourceUrl = media.source_url;
+			mediaAltText = media.alt_text;
 		}
 	}
 
+	const imagePath = getPath( mediaSourceUrl );
+	if ( imagePath ) {
+		imageFilename = last( imagePath.split( '/' ) );
+	}
+
 	return (
 		<PostFeaturedImageCheck>
 			<div className="editor-post-featured-image">
-				<MediaUploadCheck fallback={ instructions }>
-					<MediaUpload
-						title={ postLabel.featured_image || DEFAULT_FEATURE_IMAGE_LABEL }
-						onSelect={ onUpdateImage }
-						allowedTypes={ ALLOWED_MEDIA_TYPES }
-						modalClass={ ! featuredImageId ? 'editor-post-featured-image__media-modal' : 'editor-post-featured-image__media-modal' }
-						render={ ( { open } ) => (
-							<Button
-								className={ ! featuredImageId ? 'editor-post-featured-image__toggle' : 'editor-post-featured-image__preview' }
-								onClick={ open }
-								aria-label={ ! featuredImageId ? null : __( 'Edit or update the image' ) }>
-								{ !! featuredImageId && media &&
-									<ResponsiveWrapper
-										naturalWidth={ mediaWidth }
-										naturalHeight={ mediaHeight }
-									>
-										<img src={ mediaSourceUrl } alt="" />
-									</ResponsiveWrapper>
-								}
-								{ !! featuredImageId && ! media && <Spinner /> }
-								{ ! featuredImageId && ( postLabel.set_featured_image || DEFAULT_SET_FEATURE_IMAGE_LABEL ) }
-							</Button>
-						) }
-						value={ featuredImageId }
-					/>
-				</MediaUploadCheck>
+				{ !! featuredImageId &&
+					<MediaUploadCheck fallback={ instructions }>
+						<MediaUpload
+							title={ postLabel.featured_image || DEFAULT_FEATURE_IMAGE_LABEL }
+							onSelect={ onUpdateImage }
+							allowedTypes={ ALLOWED_MEDIA_TYPES }
+							modalClass="editor-post-featured-image__media-modal"
+							render={ ( { open } ) => (
+								<Button className="editor-post-featured-image__preview" onClick={ open } aria-label={ __( 'Edit or update the image' ) } aria-describedby={ `editor-post-featured-image__description-${ featuredImageId }` } >
+									{ media &&
+										<ResponsiveWrapper
+											naturalWidth={ mediaWidth }
+											naturalHeight={ mediaHeight }
+										>
+											<img src={ mediaSourceUrl } alt="" />
+										</ResponsiveWrapper>
+									}
+									{ ! media && <Spinner /> }
+								</Button>
+							) }
+							value={ featuredImageId }
+						/>
+						<p
+							id={ `editor-post-featured-image__description-${ featuredImageId }` }
+							className="screen-reader-text"
```

Comment:
Hi @Jackie6 I think here we should use the attribute hidden instead of className="screen-reader-text". The explanation of why that's the case is available at https://github.com/WordPress/gutenberg/pull/14752/files#r274131110.

Reply:
Done. Thanks for suggestions.